src/utils/adapter.ts

```diff
@@ -1,45 +1,50 @@
-import PostRobot from "post-robot";
-import { onData, onError } from "./utils";import { AxiosRequestConfig, AxiosResponse } from '../types/axios.type';
-import { ApiRequestProps, GenericObjectType } from '../types/common.types';
+import PostRobot from 'post-robot';
+import { onData, onError } from './utils';
+import { ApiRequestParams } from '../types/api.type';
+import { RequestOption, GenericObjectType } from '../types/common.types';
 
-export function createSDKAdapter(postRobot: typeof PostRobot): (config: AxiosRequestConfig) => Promise<AxiosResponse<GenericObjectType>> {
-  return async (config: AxiosRequestConfig): Promise<AxiosResponse<GenericObjectType>> => {
-    const req: ApiRequestProps = {
-      url: config?.url,
-      method: config?.method,
-      headers: config?.headers,
-      body: config?.data,
-      params: config?.params,
-    };
-
-    try {
-      const data = await dispatchPostRobotRequest(postRobot, req) as GenericObjectType;
-      return {
-        data,
-        status: data.status || 200,
-        statusText: 'OK',
-        headers: config.headers || {},
-        config,
-        request: req,
-      };
-    } catch (error) {
-      const typedError = error as GenericObjectType & { status?: number; statusText?: string; headers?: Record<string, string> };
-      return {
-        data: typedError,
-        status: typedError.status || 500,
-        statusText: typedError.statusText || 'Internal Server Error',
-        headers: typedError.headers || {},
-        config,
-        request: req,
-      };
-    }
-  };
-}
+export const dispatchPostRobotRequest = (postRobot: typeof PostRobot) => (url:string ,opts?: RequestOption): Promise<GenericObjectType> => {
+  return postRobot
+    .sendToParent("apiAdapter", {url, option:opts})
+    .then(onData)
+    .catch(onError);
+};
 
-export function dispatchPostRobotRequest(postRobot: typeof PostRobot, opts:ApiRequestProps):Promise<any> {
-    return postRobot
-      .sendToParent("apiAdapter", opts)
-      .then(onData)
-      .then((data) => data)
-      .catch(onError);
-}
\ No newline at end of file
+export const createSDKAdapter = (postRobot: typeof PostRobot) => async (config: ApiRequestParams) => {
+  try {
+    if (config.data) {
+      if (typeof config.data === "string") {
```

Comment:
I don't think this is the solution we should be implementing. I think the problem is with forwarding the request to proxy service. 
if you decided to go with this solution then try something like this.
const body = typeof config.data === 'string' ? JSON.parse(config.data): config.data;

Reply:
moving logic to AEC removing current code